libdnf/rpm/transaction.cpp

```diff
@@ -0,0 +1,746 @@
+                cb.cpio_error(item, RpmHeader(hdr));
+                break;
+            case RPMCALLBACK_SCRIPT_ERROR:
+                // amount is script tag
+                // total is return code - if (!RPMSCRIPT_FLAG_CRITICAL) return_code = RPMRC_OK
+                cb.script_error(item, RpmHeader(hdr), amount, total);
+                break;
+            case RPMCALLBACK_SCRIPT_START:
+                //amount is script tag
+                //TODO(jrohel): Define enum
+                //                 RPMTAG_PREIN; "%pre";
+                //                 RPMTAG_POSTIN; "%post";
+                //                 RPMTAG_PREUN; "%preun";
+                //                 RPMTAG_POSTUN; "%postun";
+                //                 RPMTAG_PRETRANS; "%pretrans";
+                //                 RPMTAG_POSTTRANS; "%posttrans";
+                //                 RPMTAG_VERIFYSCRIPT; "%verifyscript";
+                //                 RPMTAG_TRIGGERSCRIPTS; "%triggerprein";
+                //                 RPMTAG_TRIGGERSCRIPTS; "%triggerin";
+                //                 RPMTAG_TRIGGERSCRIPTS; "%triggerun";
+                //                 RPMTAG_TRIGGERSCRIPTS; "%triggerpostun";
+                //                 RPMTAG_FILETRIGGERSCRIPTS; "%filetriggerin";
+                //                 RPMTAG_FILETRIGGERSCRIPTS; "%filetriggerun";
+                //                 RPMTAG_FILETRIGGERSCRIPTS; "%filetriggerpostun";
+                //                 RPMTAG_TRANSFILETRIGGERSCRIPTS; "%transfiletriggerin";
+                //                 RPMTAG_TRANSFILETRIGGERSCRIPTS; "%transfiletriggerun";
+                //                 RPMTAG_TRANSFILETRIGGERSCRIPTS; "%transfiletriggerpostun";
+                cb.script_start(item, RpmHeader(hdr), amount);
+                break;
+            case RPMCALLBACK_SCRIPT_STOP:
+                // amount is script tag
+                // total is return code - if (error && !RPMSCRIPT_FLAG_CRITICAL) return_code = RPMRC_NOTFOUND
+                cb.script_stop(item, RpmHeader(hdr), amount, total);
+                break;
+            case RPMCALLBACK_INST_STOP:
+                cb.install_stop(item, RpmHeader(hdr), amount, total);
+                break;
+            case RPMCALLBACK_ELEM_PROGRESS:
+                cb.elem_progress(item, RpmHeader(hdr), amount, total);
+                break;
+            case RPMCALLBACK_VERIFY_PROGRESS:
+                cb.verify_progress(amount, total);
+                break;
+            case RPMCALLBACK_VERIFY_START:
+                cb.verify_start(total);
+                break;
+            case RPMCALLBACK_VERIFY_STOP:
+                cb.verify_stop(total);
+                break;
+            case RPMCALLBACK_UNKNOWN:
+                std::cout << "RPMCALLBACK_UNKNOWN" << std::endl;
+                break;
+        }
+
+        return rc;
+    }
+};
+
+
+Transaction::Impl::Impl(Base & base, rpmVSFlags vsflags) : base(&base) {
```

Comment:
clang-tidy complains that the constructor doesn't initialize `items`

Reply:
False positive. `items` has the default constructor.
But I will change the code to satisfy the `clang-tidy`.